omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/PatientIdentifierResource1_8.java

```diff
@@ -231,10 +231,10 @@ public PatientIdentifier save(PatientIdentifier delegate) {
 		}
 		
 		if (needToAdd) {
+			service().savePatientIdentifier(delegate);
 			delegate.getPatient().addIdentifier(delegate);
 		}
-		
-		service().savePatientIdentifier(delegate);
```

Comment:
So is what happens here in the old case is the validation runs on the "savePatientIdentifier" and it is correctly flagged as invalid, but this is only *after* the identifier is attached to the patient, so it still gets saved when the patient is saved?  Bad bug and nice find!

Reply:
Yes.  Weird one because, the patient isn't visibly/obviously saved, except by hibernate behind the scene despite the validation exception.